freshmaker/image_verifier.py

```diff
@@ -22,120 +22,119 @@
-                "$and": [
-                    {
-                        "field": "repository",
-                        "op": "=",
-                        "rvalue": repo_name
-                    },
-
-                ]
-            },
-            "projection": [
-                {"field": "*", "include": True, "recursive": True}
-            ]
-        }
+        repos = self.pgql.find_repositories_by_repository_name(repo_name)
 
-        repos = self.lb.find_container_repositories(query, auto_rebuild=False)
         if not repos:
-            raise ValueError("Cannot get repository %s from Lightblue." % repo_name)
+            raise ValueError("Cannot get repository %s from Pyxis." % repo_name)
+
         if len(repos) != 1:
-            raise ValueError("Multiple records found in Lightblue for repository %s." % repo_name)
+            raise ValueError(
+                "Multiple records found in Pyxis for image repository %s." % repo_name
+            )
 
         return repos[0]
 
-    def _get_repository_from_image(self, nvr):
+    def _get_repository_from_image(self, image):
         """
-        Returns the ContainerRepository object based on the image NVR.
+        Returns the ContainerRepository object based on the image defined by the image NVR.
         """
-        query = {
-            "objectType": "containerRepository",
-            "query": {
-                "$and": [
-                    {
-                        "field": "images.*.brew.build",
-                        "op": "=",
-                        "rvalue": nvr
-                    },
-
-                ]
-            },
-            "projection": [
-                {"field": "*", "include": True, "recursive": True}
-            ]
-        }
+        nvr = image["brew"]["build"]
 
-        repos = self.lb.find_container_repositories(query)
-        if not repos:
-            raise ValueError("Cannot get repository for image %s from Lightblue." % nvr)
-        if len(repos) != 1:
+        if "repositories" not in image or not image["repositories"]:
+            raise ValueError("Cannot get repository for image %s from Pyxis." % nvr)
+
+        if len(image["repositories"]) != 1:
             raise ValueError(
-                "Image %s found in multiple repositories in Lightblue." % nvr)
+                "Image %s found in multiple repositories in Pyxis." % nvr)
 
+        image_repo = image["repositories"][0]
+
+        if "published" not in image_repo or not image_repo["published"]:
+            for exc in conf.unpublished_exceptions:
+                if image_repo["repository"] == exc["repository"]:
+                    break
+            else:
+                raise ValueError(
+                    "Only images with published repositories or unpublished exceptions can be rebuilt, but this image's repository is not published."
+                )
+
+        repos = self.pgql.find_repositories_by_repository_name(image_repo["repository"])
         return repos[0]
```

Comment:
@qixiang Is the logic to return the first repo alright?

Reply:
Yes.